### ISSUE TITLE
Destructuring function expression (without invocation) does not report

### ISSUE THREAD

**Issue by minedeljkovic** [facebook/flow#2612]
Examples:

Tuple (motivating example):
[Try](https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVUCuA7AxgFwEs5swo44AKASgC4wBtbTAWwCMBTAJwBowBnfF0LYA5gF0wAbzBcO+TF1IMAjHwDk6yQF9UuEoMYBDPm0kBeMhTDBgAgBZxMMACayOABzhd8YFoX5+EVEyHAJiUhEANzhcIyISIA)

``` js
function foo(): [number, string] { return [1, ''] }
const [a, b] = foo // should report missing function invocation
```

Ordinary array too, if elements are destructured without rest:
[Try](https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVUCuA7AxgFwEs5swo44AKASgC4wBtbTAWwCMBTAJwF0wBvMFw75MXUgwCMAGjAAmPgF9UuEgGd8jAIZ8AvGQphgwMGoAWcTDAAmQjgAc4XTS0Jq1hbAHMyOAsVJPADc4XC0iEiA)

``` js
function foo(): [number] { return [1, 2] }
const [a] = foo // should report missing function invocation
```

But with rest it reports:
[Try](https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVUCuA7AxgFwEs5swo44AKASgC4wBtbTAWwCMBTAJwF0wBvMFw75MXUgwCMAGjAAmPgF9UuEgGd8jAIayAdPrZ8AvGQphgwIRwAOcLvjVgWhNWsLYA5mRwFipdwBucLhaRCRAA)

``` js
function foo(): [number] { return [1, 2] }
const [a, ...b] = foo // reports missing function invocation
```

Function return type makes no difference:
[Try](https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVUCuA7AxgFwEs5swo44AKASgC4xtMBbAIwFMAnMAbzA7fyYOpAIxgAvqlwkAzvjABtAIYBdMAF4yFMMGBgZACziYYAEz5sADnA7ymhGTMLYA5mRwFipZwDc4uJSISIA)

``` js
function foo(): number { return 1 }
const [a] = foo // should report missing function invocation
```

**Comment by vkurchatkin** [comment]
Probably related: https://github.com/facebook/flow/issues/1512

**Comment by somewhatabstract** [comment]
Just encountered a variation of this:
https://flow.org/try/#0JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcARFDvmQNwBQdA9I3AIJyrAA2XAnnAAsIEANZwYA5PEowArlAB26ZHAKyF+YBAUA6Orm2p4s1FgAqA4AoDmcALxwAFAEoAXE5f2AfHAAKxEGBTAB4jKCtrL2dvOABvOjhEuAMlQgV7TCoYHRMsAGFkHgAjPBFHTzsfeKSauBl5dP8SIKwdSlQILgA3LEcyAEk4ABNgIY45AgIyZ3oagF8ZhKT6xVUFejm4BmY2BQgJLChBYREAGmTDWUCbcQEsVWAoIzgi2Xg8XFGsBRhCvhRUKgrhE4MB4EFhlgDCMbio5GAuK0mCwAGJcCAAdzgACsTPAJFYRMoeKD0BisDw9ClnrlWEUwngYFghhYQQ4XO5ytFKn4Ai1QjBwjYojFqklqfAANoEBQAXQyuVZNhcs2WWDkqxlG3oQA

**Comment by gkz** [comment]
Fixed in 80ca16f77ba81a3475dd1f7a9f4d9d9e3d71bca8
